qiskit_optimization/algorithms/qrao/encoding_commutation_verifier.py

```diff
@@ -0,0 +1,70 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The EncodingCommutationVerifier."""
+
+from __future__ import annotations
+
+from qiskit.primitives import BaseEstimator
+
+from qiskit_optimization.exceptions import QiskitOptimizationError
+
+from .quantum_random_access_encoding import QuantumRandomAccessEncoding
+
+
+class EncodingCommutationVerifier:
+    """Class for verifying that the relaxation commutes with the objective function."""
+
+    def __init__(self, encoding: QuantumRandomAccessEncoding, estimator: BaseEstimator):
+        """
+        Args:
+            encoding: The encoding to verify.
+            estimator: The estimator to use for the verification.
+        """
+        self._encoding = encoding
+        self._estimator = estimator
+
+    def __len__(self) -> int:
+        return 2**self._encoding.num_vars
+
+    def __iter__(self):
+        for i in range(len(self)):
+            yield self[i]
+
+    def __getitem__(self, i: int) -> tuple[str, float, float]:
+        if i not in range(len(self)):
```

Comment:
`i in range(...)` is slow.

```suggestion
        if i < 0 or i >= len(self):
```